src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -772,6 +772,18 @@ private TypeGenerationSpec GetOrAddTypeGenerationSpec(Type type, JsonSourceGener
                 {
                     classType = ClassType.Enum;
                 }
+                else if (type.GetCompatibleGenericInterface(_iasyncEnumerableOfTType) is Type iasyncEnumerableType)
+                {
+                    if ((type.GetConstructor(Type.EmptyTypes) != null || type.IsValueType) && !type.IsAbstract && !type.IsInterface)
```

Comment:
Do we have a similar check elsewhere? Maybe implement in a shared method if so.

Reply:
Done